service/src/test/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiControllerAiNotebookTest.java

```diff
@@ -116,6 +120,48 @@ public void createAiNotebookInstance_correctZone() throws Exception {
         List.of(WorkspaceConstants.Properties.DEFAULT_RESOURCE_LOCATION));
   }
 
+  @Test
+  public void createAiNotebookInstance_populates_machineType_inGetResponse() throws Exception {
+    ApiGcpAiNotebookInstanceResource notebook =
+        mockMvcUtils
+            .createAiNotebookInstance(userAccessUtils.defaultUserAuthRequest(), workspaceId, null)
+            .getAiNotebookInstance();
+    assertEquals(
+        DEFAULT_CREATED_AI_NOTEBOOK_MACHINE_TYPE, notebook.getAttributes().getMachineType());
+  }
+
+  @Test
+  public void updateAiNotebookInstance_machineTypeAndAcceleratorConfig() throws Exception {
+    ApiGcpAiNotebookInstanceResource notebook =
+        mockMvcUtils
+            .createAiNotebookInstance(userAccessUtils.defaultUserAuthRequest(), workspaceId, null)
+            .getAiNotebookInstance();
+
+    String instanceName =
+        "projects/%s/locations/%s/instances/%s"
+            .formatted(
+                notebook.getAttributes().getProjectId(),
+                notebook.getAttributes().getLocation(),
+                notebook.getAttributes().getInstanceId());
+    // Stop the notebook so the CPU and GPU can be updated.
+    crlService.getAIPlatformNotebooksCow().instances().stop(instanceName).execute();
+
+    var updatedNotebook =
+        mockMvcUtils.updateAiNotebookInstance(
+            userAccessUtils.defaultUserAuthRequest(),
+            workspaceId,
+            notebook.getMetadata().getResourceId(),
+            "n1-highmem-2",
```

Comment:
Suggest putting the test values in static finals, so you can do compares using those and not repeating the literals.